python/ray/serve/_private/router.py

```diff
@@ -65,16 +65,48 @@ async def resolve_async_tasks(self):
         scanner.clear()
 
 
-class ReplicaSet:
+class ReplicaScheduler(ABC):
+    async def assign_replica(
+        self, query: Query
+    ) -> Union[ray.ObjectRef, "ray._raylet.StreamingObjectRefGenerator"]:
+        pass
+
+    def update_running_replicas(self, running_replicas: List[RunningReplicaInfo]):
+        pass
+
+
+class RandomStreamingReplicaScheduler(ReplicaScheduler):
```

Comment:
The reason for this is we aren't able to use the same "tracker" ref solution here... going to have to rethink how we do the scheduling, maybe "power of 2" choice routing like we've discussed before

Reply:
I think we can just go for round robin to select replica in the router at the begining. 
Definitely need to understand the scenario more. qps might not be best factor to decide which replica to use for streaming. For some video/image use case, network & connection might be very important to consider.

Further idea: we can have some system level agent to expose metrics across cluster. And router can make some "smart" choice to decide which one to use.